wagtail_transfer/operations.py

```diff
@@ -56,6 +56,38 @@ def __hash__(self):
 from .streamfield import get_object_references, update_object_ids
 
 
+class Objective:
+    """
+    An objective describes a state of an individual database object that we wish to
+    reach as a result of this import, for example:
+    "page 123 must exist at the destination in its most up-to-date form".
+    The object is identified by a model and its ID on the source site, and the
+    objective type is one of:
+
+    'exists': achieved when the object exists at the destination site and is listed in
+              destination_ids_by_source
+    'updated': achieved when the object exists at the destination site, with any data updates
+               from the source site applied, and is listed in destination_ids_by_source
+    'located': achieved when the object has been confirmed to exist at the destination and
+               listed in destination_ids_by_source, OR confirmed not to exist at the
+               destination
+    """
```

Comment:
I think having an Objective class is really nice, but we check the type string a lot in code. Now we have this, might it be worth changing to use symbols for the statuses (ala Karl's PR here for workflow: https://github.com/jacobtoppm/wagtail/pull/7 )?

Reply:
Yep, true. In fact, in the next round of changes I'm dropping the `type` attribute in favour of a `must_update` flag, which simplifies things nicely :-)